metadata/params/params.go

```diff
@@ -71,7 +96,18 @@ func visitAllParams(components ast.Object) (map[string]Params, error) {
 		if err != nil {
 			return nil, err
 		}
-		params[string(*f.Id)] = p
+		var id string
+		if f.Id != nil {
+			id = string(*f.Id)
+		} else {
+			c, isStringLiteral := f.Expr1.(*ast.LiteralString)
+			if isStringLiteral {
+				id = c.Value
+			} else {
+				return nil, fmt.Errorf("Found an unsupported component type %T", f.Expr1)
+			}
+		}
+		params[id] = p
```

Comment:
I think my comments about the object fields above are probably relevant here, too.